doc/source/reference/apis/metadata.md

```diff
@@ -22,7 +26,97 @@ this feature graduates. If you have any comments or suggestion please open the i
 
 ## Model Metadata (incubating)
 
-With Seldon you can easily define metadata for your models that is compatible with [kfserving dataplane proposal](https://github.com/kubeflow/kfserving/blob/master/docs/predict-api/v2/required_api.md#model-metadata) specification:
+With Seldon you can easily add metadata to your models.
+You can add metadata using one of two formats:
+- `v1` format that closely correlate to the current structure of `SeldonMessage`
+- `v2` format that is future-proof and fully compatible with `kfserving` dataplane proposal
+
+As you will see in following sections difference is minimal and mostly relates to the input/output format.
+
+
+### V1 DataPlane
+
+In order to use `v1` metadata format you need to specify `apiVersion: v1`.
+
+#### Array input/output
+```YAML
+apiVersion: v1
+name: my-model-name
+versions: [ my-model-version-01 ]
+platform: seldon-custom
+inputs:
+  datatype: array
+  shape: [ 2, 2 ]
+outputs:
+  datatype: array
+  shape: [ 1 ]
+```
+This metadata would mean that following two inputs are valid for this model:
+```JSON
+{"data": {"names": ["input"], "ndarray": [[1, 2], [3, 4]]}}
+```
+and
+```JSON
+{"data": {"names": ["input"], "tensor": {"values": [1, 2, 3, 4], "shape": [2, 2]}}
+```
+
+#### jsonData input/output
+```YAML
+apiVersion: v1
+name: my-model-name
+versions: [ my-model-version-01 ]
+platform: seldon-custom
+inputs:
+  datatype: jsonData
+  schema:
+      type: object
+      properties:
+          my-names:
+              type: array
+              items:
+                  type: string
+          my-data:
+            type: array
+            items:
+                type: number
+                format: double
+outputs:
+  datatype: array
+  shape: [ 1 ]
+```
+
+Example model input:
+```JSON
+{"jsonData": {"my-names": ["a", "b", "c"], "my-data": [1.0, 4.2, 3.14]}}
+```
+
+The `schema` field is optional and can leaves user total freedom over its structure.
+
+Note: as you can see you can mix inputs and outputs of different types!
+
+
+#### strData input/output
+```YAML
+apiVersion: v1
+name: my-model-name
+versions: [ my-model-version-01 ]
+platform: seldon-custom
+inputs:
+  datatype: strData
```

Comment:
nit: should we just call the new datatype `str` instead of `strData`?

Reply:
Same as above.